language/stdlib/modules/LibraAccount.move

```diff
@@ -659,32 +659,34 @@ module LibraAccount {
     }
 
     public fun add_tier(blessed: &signer, addr: address, tier_upperbound: u64) acquires Role {
-        DesignatedDealer::assert_account_is_blessed(blessed);
+        Association::assert_account_is_blessed(blessed);
         let dealer =
             &mut borrow_global_mut<Role<DesignatedDealer::Dealer>>(addr).role_data;
         DesignatedDealer::add_tier(dealer, tier_upperbound)
     }
 
     public fun update_tier(blessed: &signer, addr: address, tier_index: u64, new_upperbound: u64) acquires Role {
-        DesignatedDealer::assert_account_is_blessed(blessed);
+        Association::assert_account_is_blessed(blessed);
         let dealer =
             &mut borrow_global_mut<Role<DesignatedDealer::Dealer>>(addr).role_data;
         DesignatedDealer::update_tier(dealer, tier_index, new_upperbound)
     }
 
     /// Create a designated dealer account at `new_account_address` with authentication key
-    /// `auth_key_prefix` | `new_account_address`, for non synthetic CoinType
+    /// `auth_key_prefix` | `new_account_address`, for non synthetic CoinType.
+    /// Creates Preburn resource under account 'new_account_address'
     public fun create_designated_dealer<CoinType>(
         blessed: &signer,
         new_account_address: address,
         auth_key_prefix: vector<u8>,
     ) {
-        DesignatedDealer::assert_account_is_blessed(blessed);
+        Association::assert_account_is_blessed(blessed);
         Transaction::assert(!Libra::is_synthetic_currency<CoinType>(), 202);
         let new_dd_account = create_signer(new_account_address);
+        Event::publish_generator(&new_dd_account);
+        Libra::publish_preburn_to_account<CoinType>(blessed, &new_dd_account);
```

Comment:
As a clarification, is there a distinct designated dealer account for each currency type? Or could a given designated dealer address hold multiple preburn resources (one for each currency).

Reply:
There will be a singular DD per currency. For MVP we don't need to link them together (alla Parent/child VASP)

Reply:
Awesome!